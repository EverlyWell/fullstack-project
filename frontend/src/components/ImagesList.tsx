import React from 'react';
import {ImageListProps} from "../types";
import ImageCard from "./ImageCard";

const ImagesList: React.FunctionComponent<ImageListProps> = ({images}) => {
    const listItems = images.map((image) => <ImageCard key={image.id} image={image}/>);

    return(
        <div className={'ImageList'}>
            {listItems}
        </div>
    )
}
export default ImagesList