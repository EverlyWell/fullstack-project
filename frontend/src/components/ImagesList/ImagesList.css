ul {
  columns: 6 200px;
  column-gap: 1rem;
}



ul li {
    list-style: none;
    text-align: center;
    width: 150px;
    background: #EC985A;
    color: white;
    margin: 0 1rem 1rem 0;
    display: inline-block;
    width: 100%;
    text-align: center;
    font-family: system-ui;
    font-weight: 900;
    font-size: 1rem;
}

@media screen and (min-width: 37.5rem) {
  :root {
    --gridList-column: 180px;
    --gridList-rows: 200px;
  }
}

@media screen and (min-width: 60rem) {
  :root {
    --gridList-column: 300px;
    --gridList-rows: 210px;
  }
}
