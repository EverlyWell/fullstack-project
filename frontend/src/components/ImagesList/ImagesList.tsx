import React from 'react'

import Image from '../../components/Image'

import './ImagesList.css'

export default function ImagesList(props: {images: Array<{ title: string, imageUrl: string, id: string }>,
    loading: boolean
    }) {
    const { images, loading } = props;

    if(loading) {
        return (
            <div>
                cargando...
            </div>
        )
    }

    return (
        <ul className='imagesList'>
            {
                images.map(({title, imageUrl, id}) =>
                <li key={id}>
                    <Image
                        key={id}
                        title={title}
                        imageUrl={imageUrl}
                        className={'Image'}
                    />
                </li>
                )
            }
        </ul>
    )
}
