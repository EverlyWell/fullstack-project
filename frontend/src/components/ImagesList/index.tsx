import ImagesList from './ImagesList';
export default ImagesList;
