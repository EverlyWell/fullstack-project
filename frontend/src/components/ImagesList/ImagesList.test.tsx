
import React from 'react';
import { shallow } from 'enzyme';

import ImagesList from '../ImagesList';

describe('ImagesList', () => {
  let tree, props;

  const buildTree = () => {
      const props = {
          images: [
              {
                  imageUrl: 'https://picsum.photos/300/200',
                  title: 'photo 1',
                  id: '189094',
              },
              {
                  imageUrl: 'https://picsum.photos/200/300',
                  title: 'photo 2',
                  id: '189093',
              },
          ],
          loading: false,
      }

      return shallow(<ImagesList {...props} />);
  };

    it('matches the snapshot', () => {
        tree = buildTree();
        expect(tree).toMatchSnapshot();
    });
});
