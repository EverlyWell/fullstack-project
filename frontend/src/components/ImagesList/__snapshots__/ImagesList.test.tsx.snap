// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagesList matches the snapshot 1`] = `
<ul
  className="imagesList"
>
  <li
    key="189094"
  >
    <ImagesContainer
      className="Image"
      id="189094"
      imageUrl="https://picsum.photos/300/200"
      key="189094"
      title="photo 1"
    />
  </li>
  <li
    key="189093"
  >
    <ImagesContainer
      className="Image"
      id="189093"
      imageUrl="https://picsum.photos/200/300"
      key="189093"
      title="photo 2"
    />
  </li>
</ul>
`;
